### DISCUSSION TITLE
Could not find the task 'build'. VS MAC version C

### DISCUSSION THREAD

**Discussion by Superice2311** [microsoft/vscode-cpptools#11439] [Q&A]
hi everyone, i am a student that learning c++, after i install vs mac i find that i cannot run and debug which shows 'cannot build and debug because the active file is not a c or c++ source file.' Then i find this page and downloaded a vsix file(cpptools-osx-arm64) .After that the problem then is change to "could not find the task 'build'." Thank for help.

**Comment by michelleangela** [comment]
@Superice2311 
Could you provide more info on how you tried to build and debug? Based on the error ` cannot build and debug because the active file is not a c or c++ source file.` it could be the active file wasn't a valid source file. When you do a build and debug using for example the `clang++ build and debug active file` command, the active file (e.g. file currently in view and selected) needs to be the source file you're trying to build and debug.

The extension also has this getting started tutorial for setting up the C/C++ extension and using the compiler Clang on macOS.
https://code.visualstudio.com/docs/cpp/config-clang-mac
